### ISSUE TITLE
Build process fails for v0.4 / gummy-git AUR

### ISSUE THREAD

**Issue by mi544** [Fushko/gummy#20]
Hi.

Following the build instructions for v0.4, the `cmake --build` terminates early with the following error:
`gummy/gummyd/gummyd/sysfs_devices.cpp:27:43: error: variable ‘constexpr const std::array<std::basic_string_view<char>, 2> gummyd::sysfs::constants::als::lux_filenames’ has initializer but incomplete type`
I am building against the latest commit fbf2948bdba4340671fd5acb3cae92d35330d729 which is what v0.4 release is pointed at

Since this may affect other people, I'll keep the AUR `gummy` package at v0.3 for now. That way it still correctly installs and functions for Arch users. I already updated the `gummy-git` [PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=gummy-git) to include new dependencies and build steps though.

Right now the `gummy` AUR package is still at v0.3 and functions properly, but `gummy-git` breaks for me in the same way

Let me know if any more information is needed!
<details>
<summary><h3>Full build log</h3></summary>

```
$ git clone https://github.com/Fushko/gummy.git --depth 1 && cd gummy
Cloning into 'gummy'...
remote: Enumerating objects: 50, done.
remote: Counting objects: 100% (50/50), done.
remote: Compressing objects: 100% (46/46), done.
remote: Total 50 (delta 2), reused 18 (delta 1), pack-reused 0
Receiving objects: 100% (50/50), 44.69 KiB | 2.63 MiB/s, done.
Resolving deltas: 100% (2/2), done.


$ mkdir build && cd build


$ cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE="Release"
-- The CXX compiler identification is GNU 12.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) at /usr/share/cmake/Modules/FetchContent.cmake:1284 (message):
  The DOWNLOAD_EXTRACT_TIMESTAMP option was not given and policy CMP0135 is
  not set.  The policy's OLD behavior will be used.  When using a URL
  download, the timestamps of extracted files should preferably be that of
  the time of extraction, otherwise code that depends on the extracted
  contents might not be rebuilt if the URL changes.  The OLD behavior
  preserves the timestamps from the archive instead, but this is usually not
  what you want.  Update your project to the NEW behavior or specify the
  DOWNLOAD_EXTRACT_TIMESTAMP option with a value of true to avoid this
  robustness issue.
Call Stack (most recent call first):
  CMakeLists.txt:10 (FetchContent_Declare)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/share/cmake/Modules/FetchContent.cmake:1284 (message):
  The DOWNLOAD_EXTRACT_TIMESTAMP option was not given and policy CMP0135 is
  not set.  The policy's OLD behavior will be used.  When using a URL
  download, the timestamps of extracted files should preferably be that of
  the time of extraction, otherwise code that depends on the extracted
  contents might not be rebuilt if the URL changes.  The OLD behavior
  preserves the timestamps from the archive instead, but this is usually not
  what you want.  Update your project to the NEW behavior or specify the
  DOWNLOAD_EXTRACT_TIMESTAMP option with a value of true to avoid this
  robustness issue.
Call Stack (most recent call first):
  CMakeLists.txt:13 (FetchContent_Declare)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/share/cmake/Modules/FetchContent.cmake:1284 (message):
  The DOWNLOAD_EXTRACT_TIMESTAMP option was not given and policy CMP0135 is
  not set.  The policy's OLD behavior will be used.  When using a URL
  download, the timestamps of extracted files should preferably be that of
  the time of extraction, otherwise code that depends on the extracted
  contents might not be rebuilt if the URL changes.  The OLD behavior
  preserves the timestamps from the archive instead, but this is usually not
  what you want.  Update your project to the NEW behavior or specify the
  DOWNLOAD_EXTRACT_TIMESTAMP option with a value of true to avoid this
  robustness issue.
Call Stack (most recent call first):
  CMakeLists.txt:16 (FetchContent_Declare)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Using the multi-header code from /home/mi/testing/gummy/build/_deps/json-src/include/
-- Implicit conversions are disabled
-- Module support is disabled.
-- Version: 9.1.0
-- Build type: Release
-- CXX_STANDARD: 11
-- Performing Test has_std_11_flag
-- Performing Test has_std_11_flag - Success
-- Performing Test has_std_0x_flag
-- Performing Test has_std_0x_flag - Success
-- Required features: cxx_variadic_templates
-- Build spdlog: 1.11.0
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Build type: Release
-- Configuring done
-- Generating done
-- Build files have been written to: /home/mi/testing/gummy/build


$ cmake --build . && sudo cmake --install .
[  2%] Building CXX object _deps/fmt-build/CMakeFiles/fmt.dir/src/format.cc.o
[  5%] Building CXX object _deps/fmt-build/CMakeFiles/fmt.dir/src/os.cc.o
[  8%] Linking CXX static library libfmt.a
[  8%] Built target fmt
[ 11%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/spdlog.cpp.o
[ 14%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/stdout_sinks.cpp.o
[ 17%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/color_sinks.cpp.o
[ 20%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/file_sinks.cpp.o
[ 22%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/async.cpp.o
[ 25%] Building CXX object _deps/spdlog-build/CMakeFiles/spdlog.dir/src/cfg.cpp.o
[ 28%] Linking CXX static library libspdlog.a
[ 28%] Built target spdlog
[ 31%] Building CXX object _deps/cli11-build/src/CMakeFiles/CLI11.dir/Precompile.cpp.o
[ 34%] Linking CXX static library libCLI11.a
[ 34%] Built target CLI11
[ 37%] Building CXX object gummy/gummyd/CMakeFiles/libgummyd.dir/gummyd/api.cpp.o
[ 40%] Building CXX object gummy/gummyd/CMakeFiles/libgummyd.dir/gummyd/file.cpp.o
[ 42%] Building CXX object gummy/gummyd/CMakeFiles/libgummyd.dir/gummyd/constants.cpp.o
[ 45%] Building CXX object gummy/gummyd/CMakeFiles/libgummyd.dir/gummyd/utils.cpp.o
[ 48%] Linking CXX static library libgummyd.a
[ 48%] Built target libgummyd
[ 51%] Building CXX object gummy/gummyd/gummyd/CMakeFiles/gummyd.dir/gamma.cpp.o
[ 54%] Building CXX object gummy/gummyd/gummyd/CMakeFiles/gummyd.dir/display.cpp.o
[ 57%] Building CXX object gummy/gummyd/gummyd/CMakeFiles/gummyd.dir/sysfs.cpp.o
[ 60%] Building CXX object gummy/gummyd/gummyd/CMakeFiles/gummyd.dir/sysfs_devices.cpp.o
/home/mi/testing/gummy/gummy/gummyd/gummyd/sysfs_devices.cpp:27:43: error: variable ‘constexpr const std::array<std::basic_string_view<char>, 2> gummyd::sysfs::constants::als::lux_filenames’ has initializer but incomplete type
   27 | constexpr std::array<std::string_view, 2> lux_filenames = {
      |                                           ^~~~~~~~~~~~~
make[2]: *** [gummy/gummyd/gummyd/CMakeFiles/gummyd.dir/build.make:118: gummy/gummyd/gummyd/CMakeFiles/gummyd.dir/sysfs_devices.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:341: gummy/gummyd/gummyd/CMakeFiles/gummyd.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

</details>

**Comment by Fushko** [comment]
Hi, thank you for posting. I pushed a speculative fix to the master branch. Please let me know if you still have issues.


The CMake warnings are due to this [new policy](https://cmake.org/cmake/help/latest/policy/CMP0135.html) in CMake 3.24+ when fetching external projects, which didn't appear on my system with 3.22.1.

These can be safely ignored because these URLs are most definitely not going to change. I´d rather not have any warnings though, so I'll work on a solution at some point.

**Comment by Fushko** [comment]
Commit 0f59a3a in master should fix it. Can you check?

**Comment by mi544** [comment]
Just checked, everything seems to function as it should now! I can build, install gummy, and set the screen temperature fine.
Would the changes be released in a new version or will 0.4 be updated instead?

**Comment by Fushko** [comment]
> Would the changes be released in a new version or will 0.4 be updated instead?

I will release an hotfix once #20 is sorted out.

**Comment by Fushko** [comment]
Fixed in 0.4.1.
